src/sentry/api/endpoints/relay_register.py

```diff
@@ -184,8 +184,15 @@ def post(self, request):
                 relay_id=relay_id, public_key=public_key, is_internal=is_internal
             )
         else:
-            relay.last_seen = timezone.now()
             relay.is_internal = is_internal
             relay.save()
 
+        try:
+            relay_usage = RelayUsage.objects.get(relay_id=relay_id, version=version)
```

Comment:
With your migration being marked as dangerous, there will be some time between the deploy happening and the migration being manually run. Will this code path get run in that time window?

Reply:
The migrations need to run before the deploy. The old code may run after the migrations but the new code should **NOT** run before migrations. 
I thought the migrations are run before the code deployment.

Reply:
Non-dangerous migrations are run before deploys. Dangerous migrations are run after. I would split your migration into safe and dangerous parts. Ship the new model, table, and make the old columns nullable, and remove the columns from the django model state. Then in a separate deploy do your 'dangerous' column drops.

Reply:
got it, thanks